label_studio/ml_models/api.py

```diff
@@ -69,17 +70,49 @@ class ModelInterfaceAPI(viewsets.ModelViewSet):
         PUT=all_permissions.models_change,
         POST=all_permissions.models_create,
     )
+    def get_serializer_class(self):
+        if self.action == 'create':
+            return ModelInterfaceCreateSerializer
+        return ModelInterfaceSerializer
 
     def get_queryset(self):
         return ModelInterface.objects.filter(organization_id=self.request.user.active_organization_id)
+    
+    def create(self, request, *args, **kwargs):
+        data = self.request.data
+        data['organization']=self.request.user.active_organization_id
+        serializer = self.get_serializer(data=data)
+        self.perform_create(serializer)
+        return Response(serializer.data,status=201)
 
     def perform_create(self, serializer):
+        associated_projects_data = self.request.data.get('associated_projects',[])
         serializer.is_valid(raise_exception=True)
 
         # we need to save these fields for faster access and filters without excess joins
         serializer.validated_data['organization'] = self.request.user.active_organization
         serializer.validated_data['created_by'] = self.request.user
         serializer.save()
+        instance = serializer.instance
+       
+        model_interface = ModelInterface.objects.filter(pk=instance.pk)[0]
+        for id in associated_projects_data:
```

Comment:
nit: if this is just a list of ids, mind if we name the var `associated_project_ids`?

Reply:
okay

Reply:
done